cmd/agent/common/helpers.go

```diff
@@ -86,19 +86,19 @@ func SelectedCheckMatcherBuilder(checkNames []string, minInstances uint) func(co
 // SetupInternalProfiling is a common helper to configure runtime settings for internal profiling.
 func SetupInternalProfiling(cfg config.ConfigReader, configPrefix string) {
 	if v := cfg.GetInt(configPrefix + "internal_profiling.block_profile_rate"); v > 0 {
-		if err := settings.SetRuntimeSetting("runtime_block_profile_rate", v); err != nil {
+		if err := settings.SetRuntimeSetting("runtime_block_profile_rate", v, settings.SourceDefault); err != nil {
```

Comment:
Follow-up on https://github.com/DataDog/datadog-agent/pull/18295#discussion_r1276483445

Since this settings is pulled from the configuration it might not be the default but the user set value. I think a `settins.SourceConfig` would be interesting to have.